cypress/integration/utils/image-segmentation-editor-test-part/LeftSideBar.js

```diff
@@ -0,0 +1,148 @@
+
+    cy.get("div[id='root']").trigger("mousemove", 200, 200, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 200, 200, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 200, 200, { button: 0 })
+
+    cy.get("div[id='root']").trigger("mousemove", 200, 100, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 200, 100, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 200, 100, { button: 0 })
+
+    cy.get("div[id='root']").trigger("mousemove", 150, 80, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 150, 80, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 150, 80, { button: 0 })
+
+    cy.log("open polygon detail box")
+    cy.get(
+      "div[class='MuiPaper-root makeStyles-regionInfo-159 highlighted MuiPaper-elevation1 MuiPaper-rounded']"
+    ).click()
+    cy.contains("polygon")
+
+    cy.log("Move polygon point")
+    cy.get("div[id='root']").trigger("mousemove", 150, 80, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 150, 80, { button: 0 })
+    cy.get("div[id='root']").trigger("mousemove", 300, 80, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 300, 80, { button: 0 })
+
+    cy.log("Add a new polygon point")
+    cy.get("div[id='root']").trigger("mousemove", 250, 140, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 250, 140, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 250, 140, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 250, 140, { button: 0 })
+    cy.get("div[id='root']").trigger("mousemove", 450, 140, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 450, 140, { button: 0 })
+
+    cy.matchImageSnapshot("PolygonCreated", {
+      failureThreshold: 0.01,
+    })
+  })
+  it("Add expanding line", () => {
+    goToEditor("Image Segmentation", 0)
+    clickOnLeftSidebar("Add Expanding Line")
+    cy.get("div[id='root']").trigger("mousemove", 400, 400, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 400, 400, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 400, 400, { button: 0 })
+
+    cy.get("div[id='root']").trigger("mousemove", 400, 500, { button: 0 })
+    cy.get("div[id='root']").trigger("mousedown", 400, 500, { button: 0 })
+    cy.get("div[id='root']").trigger("mousemove", 500, 500, { button: 0 })
+    cy.get("div[id='root']").trigger("mouseup", 500, 500, { button: 0 })
+
+    cy.get("div[id='root']").trigger("keydown", 400, 500, {
+      code: 27,
+      key: "Escape",
+      keyCode: 27,
+    })
+    cy.get("div[id='root']").trigger("keypress", 400, 500, {
+      code: 27,
+      key: "Escape",
+      keyCode: 27,
+    })
+    cy.get("div[id='root']").trigger("keyup", 400, 500, {
+      code: 27,
+      key: "Escape",
+      keyCode: 27,
+    })
+
+    cy.get(
+      "div[class='MuiPaper-root makeStyles-regionInfo-160 highlighted MuiPaper-elevation1 MuiPaper-rounded']"
```

Comment:
this is most likely too tightly coupled to the implementation, when makeStyles changes (it will if we introduce more classes) this test will break

Reply:
Yeah i would put id in place the issue is opened : https://github.com/UniversalDataTool/universal-data-tool/issues/497